src/__tests__/union.test.ts

```diff
@@ -229,6 +276,71 @@ describe('union', () => {
         });
       });
 
+      describe('readable/writeable', () => {
+        it('writes to the last vol added', done => {
+          const vol1 = Volume.fromJSON({});
+          const vol2 = Volume.fromJSON({});
+          const ufs = new Union();
+          ufs.use(vol1 as any).use(vol2 as any);
+          ufs.writeFile('/foo', 'bar', (err) => {
+            vol2.readFile('/foo', 'utf8', (err, res) => {
+              expect(res).toEqual('bar');
+              done();
+            });
+          });
+        });
+
+        it('writes to the latest added volume without writeable=false', done => {
+          const vol1 = Volume.fromJSON({});
+          const vol2 = Volume.fromJSON({});
+          const ufs = new Union();
+          ufs.use(vol1 as any).use(vol2 as any, { writeable: false });
+          ufs.writeFile('/foo', 'bar', (err) => {
+            vol1.readFile('/foo', 'utf8', (err, res) => {
+              expect(res).toEqual('bar');
+              done();
+            });
+          });
+        });
+
+        it('writes to the latest added writeable vol', done => {
+          const vol1 = Volume.fromJSON({});
+          const vol2 = Volume.fromJSON({});
+          const ufs = new Union();
+          ufs.use(vol1 as any).use(vol2 as any, { readable: false });
+          ufs.writeFile('/foo', 'bar', (err) => {
+            vol2.readFile('/foo', 'utf8', (err, res) => {
+              expect(res).toEqual('bar');
+              done();
+            });
+          });
+        });
+
+        it('not throw error if write operation attempted with all volumes writeable=false', done => {
```

Comment:
Generally the default behaviour in file systems is to throw if you cannot write a file - that's why we have things like `/dev/null` which provide a way to write a file that will always succeed in the rare situations that we don't care what happens.

I think this behaviour should be inverted.

Additionally, you could make these test cases more readable by using `describe`, and structuring the title to start with the `it`, i.e

```
describe('when all volumes have writable as false', () => {
  it('throws an error when writing', done => {
    /* ... /*
  });
});
```

Reply:
OK yes i see your point here. I'm going to implement it such that if the last volume throws a `SkipMethodError` we can inform them with a specific error message. Does that sound ok?

Reply:
Awesome; You shouldn't even need to throw a special error, as just throwing an error in `noop` should result in behaviour that matches what would happen if the "real" `fs` throws an error.

Reply:
I meant from a user perspective - surely they'd like to know they've configured a stack of volumes incorrectly?

Reply:
I think that the additional complexity that that would introduce isn't worth it for now.

You're already making assumptions by saying that they must have configured the stack incorrectly: why is it incorrect to have not allow reading or writing to all of the volumes?

From there, the argument could be made if we're going to throw something special based on that conditional ideally we should do it asap - so why add checks when calling `use`? etc etc.

If it turns out that this is a common problem then it'll be easy to implement this in a follow up PR, as nothing we're doing in this one will make it harder to support.

-----

However we should get this mostly for free already, by the fact that we generally re-throw the last error; so now we come full circle back to [one of my original comments](https://github.com/streamich/unionfs/pull/441#discussion_r401204752) 😉